### DISCUSSION TITLE
sea-orm-rocket temporary fork for Rocket 0.6 compatibility

### DISCUSSION THREAD

**Discussion by amyipdev** [SeaQL/sea-orm#2272] [General]
On a new project I started, I decided to switch to the development channel of Rocket, giving me access to the v0.6 functionality.

While I'd normally be fine on stable, a key feature - HTTP/3 support - is not going to be backported, and with all the other changes coming in v0.6, I decided to just switch over now to avoid porting issues later.

However, upon adding SeaORM in, I had unexplained compiler errors - until I dug further. SeaORM is pegged to v0.5.1 and cannot, even with `crates.io` patches, be overridden past that. Not that it would work anyways, as it uses an `info_!` macro that no longer exists in v0.6.

For now, I've created a minimal patch fork, which overrides Rocket to a particular mainline revision: https://github.com/amyipdev/sea-orm/commit/48aa4ced8cb54611c42bde53f4f6b77e3fe20c74

And for anyone who needs it, to import it in Cargo.toml:
```toml
[dependencies.sea-orm-rocket]
git = "https://github.com/amyipdev/sea-orm"
branch = "rocket-0.6.0-dev"
```

**Comment by tyt2y3** [comment]
I think we can start migrating to 0.6
Can you open a PR on this repo, bumping everything?
https://crates.io/crates/sea-orm-rocket/versions
I think old versions are published, so we can move forward without breaking anyone.

**Reply by amyipdev** [reply]
Problem is that at the moment, there is no `v0.6.0` on Crates. I can start getting something drafted but it wouldn't be remotely ready for review until `v0.6.0` is actually published, or at least `v0.6.0-rc.1`.
